### PR TITLE
Fix get not valid number at expression estimate

### PR DESCRIPTION
**Pull request by Youngwb**
#1835 
At compute expression statistics, we may get NaN min/max value because of the null value，we need to adjust it before compute predicate factor.
A FoldConstantRule problem was discovered during the revision process, which was modified along with it

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/FoldConstantsRule.java [hunk 1]
```diff
@@ -191,10 +191,14 @@ public ScalarOperator visitLikePredicateOperator(LikePredicateOperator predicate
...
 
-    private boolean notAllConstant(List<ScalarOperator> operators) {
+    private boolean notAllConstantRef(List<ScalarOperator> operators) {
         return !operators.stream().allMatch(ScalarOperator::isConstantRef);
     }
 
+    private boolean notAllConstant(List<ScalarOperator> operators) {
+        return !operators.stream().allMatch(ScalarOperator::isConstant);
+    }
+
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/FoldConstantsRule.java:201]:
```suggestion

```

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/FoldConstantsRule.java [hunk 2]
```diff
@@ -191,7 +191,7 @@ public ScalarOperator visitLikePredicateOperator(LikePredicateOperator predicate
         return predicate;
     }
 
-    private boolean notAllConstant(List<ScalarOperator> operators) {
+    private boolean notAllConstantRef(List<ScalarOperator> operators) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/FoldConstantsRule.java:194]: Why rename this? constant no ref meaning.

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/FoldConstantsRule.java:194]: It corresponds to ScalarOperator::isConstantRef, we need use ScalarOperator::isConstant in some case later.

Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/FoldConstantsRule.java:194]: I think `ScalarOperator::isConstantRef` also should rename to isConstantOperator, or directly check `instanceof ConstantOperator`.   The key point is constant no ref meaning.

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/FoldConstantsRule.java]: Done

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**